parallel_algorithms/P3732/numeric_ranges_algorithms.md

```diff
@@ -526,6 +561,55 @@ reduction algorithms to have projections.
 `ranges::transform_{in,ex}clusive_scan` as well as `ranges::{in,ex}clusive_scan`, and do not provide projections for any of
 them.
 
+### `reduce_into` and `transform_reduce_into`
+
+We propose new algorithms `reduce_into` and `transform_reduce_into`.
+These work like `reduce` and `transform_reduce`,
+except that instead of returning the reduction result by value,
+they write it to the first element of an output range.
+
+The `reduce_into` algorithm has
+[precedent in the Thrust library](https://nvidia.github.io/cccl/thrust/api_docs/algorithms/reductions.html).
+Its performance advantange is that the algorithm can write its result
+directly to special memory associated with parallel execution,
+such as accelerator memory or a NUMA (Non-Uniform Memory Access) domain
+where the algorithm's threads run.
+
+#### Output should be a single iterator

```

Comment:
Advantage of a sized output range is that it can have a hardened precondition.

It doesn't need random access.  It only needs to be a forward range for the parallel algorithms.

Reply:
Added discussion of this in PR #36.